README.md

```diff
@@ -1,63 +1,109 @@
+
+    -   replace `dredd-hooks-{{mylanguage}}` with a path to your hooks handler executable which you want to get tested (e.g. `./bin/dredd_hooks`)
+    -   replace `{{myextension}}` by the extension of the hooks handler language (e.g. `.py`),
+    -   uncomment the code blocks and rewrite them to the hooks handler language.
+
+Now you have the test suite ready.
+
+### Usage
 
-### Enable the features for your language
+Every time you run `npx dredd-hooks-template test` (or `npm test`), you should see the test suite running. The end goal is that all the tests pass:
 
-1. Open the feature files in `./features/*.feature`
-1. In all of them, replace:
-  - `{{mylanguage}}` by the hooks handler command for you language
-  - `{{myextension}}` by the extension for your language
-1. Implement the code examples in your language
-1. Run the test suite: `bundle exec cucumber`
+![test suite passing](passing.png)
+
+You should add the `package.json` file to Git. When starting from scratch, you can run `npm install` to install the JavaScript dependencies.
+
+<a name="upgrading"></a>
+
+### Upgrading
+
+Watch for newer versions of the [dredd-hooks-template package](https://www.npmjs.com/package/dredd-hooks-template) and upgrade regularly to keep up with development of Dredd and the test suite itself. To upgrade, run:
+
+```
+$ npx dredd-hooks-template upgrade
+```
 
-### Add the features to your project
+It upgrades the package to the latest version and copies the latest feature files to the project's `./features/` directory. It won't overwrite the existing files as the names of the new files get suffixed with version. Then it's up to you to compare the old and new files, spot changes, and update the project's test suite.
 
-If the test suite did run as expected, you can now add the features to your project.
-To do so, copy to your project:
+### Reference Implementations
 
-1. the entire `features/` directory
-1. the `Gemfile`, `Gemfile.lock` and `.ruby-version`
+The [Python hooks](https://github.com/apiaryio/dredd-hooks-python) and the [Ruby hooks](https://github.com/apiaryio/dredd-hooks-ruby) can be used as examples of how to use this cross-language test suite.
 
-Your should now be able to install the features dependencies and run the test suite in your project.
+### Keep Tests Running with CI
 
-Finally, make `bundle exec cucumber` part of your test suite and CI (see `.travis.example.yml` if you are using [Travis CI][travis]).
+To make sure the hooks handler will always work correctly with Dredd and the expectations won't get accidentally broken, put the tests into [Travis CI](https://travis-ci.org), which runs the tests for each change on your repository. See existing configuration files for inspiration:
 
-  [travis]: https://travis-ci.org
+- Python: [.travis.yml](https://github.com/apiaryio/dredd-hooks-python/blob/master/.travis.yml)
+- Ruby: [.travis.yml](https://github.com/apiaryio/dredd-hooks-ruby/blob/master/.travis.yml)
 
 ## Development
```

Comment:
Depending on the likelihood of contribution and the package's intent, we could also move this section to a designated `CONTRIBUTING.md`.